tests/MetricsBuilderTest.php

```diff
@@ -169,4 +169,31 @@ public function resolveAndAppendFromArgsProvider(): array
             ]
         ];
     }
+
+    /**
+     * Tests that metric middlewares are appended just once.
+     *
+     * @return void
+     */
+    public function testAppendMetricsCaptureMiddlewareJustOnce(): void {
+        $handlerList = new HandlerList(function (){});
```

Comment:
I think there should be an assertion that "appendBuild" is only called once.  Prior to this change, I'm pretty sure the second attempt to append it would just override the first one since they're both in the build step

Reply:
Actually it will not override it:
```
// The middleware will be added not matter the name
if ($prepend) {
 $this->steps[$step][] = [$middleware, $name];
} else {
    array_unshift($this->steps[$step], [$middleware, $name]);
}

// Here will be overridden, however, the middleware is already added.
if ($name) {
    $this->named[$name] = $step;
}
```